app/src/main/java/org/simple/clinic/medicalhistory/newentry/NewMedicalHistoryUiRenderer.kt

```diff
@@ -36,5 +36,9 @@ class NewMedicalHistoryUiRenderer(
     }
 
     ui.showDiagnosisRequiredError(model.showDiagnosisRequiredError)
+
+    if (model.buttonState == ButtonState.SAVING) {
```

Comment:
Can we extract this check into a computed property on `NewMedcalHistoryModel`, say `isSaving` or something?